src/main/java/persistence/sql/dml/impl/SelectQueryBuilder.java

```diff
@@ -0,0 +1,48 @@
+package persistence.sql.dml.impl;
+
+import persistence.sql.clause.Clause;
+import persistence.sql.clause.ConditionalClause;
+import persistence.sql.QueryBuilder;
+import persistence.sql.common.util.NameConverter;
+import persistence.sql.data.QueryType;
+import persistence.sql.dml.MetadataLoader;
+
+import java.util.Arrays;
+
+public class SelectQueryBuilder implements QueryBuilder {
+    private final NameConverter nameConverter;
+
+    public SelectQueryBuilder(NameConverter nameConverter) {
+        this.nameConverter = nameConverter;
+    }
+
+    @Override
+    public QueryType queryType() {
+        return QueryType.SELECT;
+    }
+
+    @Override
+    public boolean supported(QueryType queryType) {
+        return QueryType.SELECT.equals(queryType);
+    }
+
+    @Override
+    public String build(MetadataLoader<?> loader, Clause... clauses) {
+        String columns = String.join(DELIMITER, loader.getColumnNameAll(nameConverter));
+        String tableName = loader.getTableName();
+
+        StringBuilder query = new StringBuilder("SELECT %s FROM %s".formatted(columns, tableName));
+
+        ConditionalClause[] conditionalClauses = Arrays.stream(clauses)
+                .filter(clause -> clause instanceof ConditionalClause)
+                .map(clause -> (ConditionalClause) clause)
+                .toArray(ConditionalClause[]::new);
```

Comment:
```suggestion
      ConditionalClause[] conditionalClauses = Arrays.stream(clauses)
                .filter(clause -> clause instanceof ConditionalClause)
                .map(clause -> (ConditionalClause) clause)
                .toArray(ConditionalClause[]::new);
```
쿼리빌더들에서 필터링과정이 모두 존재하네요 ㅜ
작동되는데에 문제는 없겠지만, `Clause` 추상화 의미가 퇴색되는것 같긴하네요 🤔

Reply:
`UpdateQueryBuilder`에 
```java
private String getWhereClause(Clause[] clauses) {
        ConditionalClause[] conditionalClauses = Arrays.stream(clauses)
                .filter(clause -> clause instanceof ConditionalClause)
                .map(clause -> (ConditionalClause) clause)
                .toArray(ConditionalClause[]::new);

        return getWhereClause(conditionalClauses);
    }
```
같은 코드가 존재하는데 어떻게 재활용을 할 수 있을지 혹은 재활용을 해야할지 고민이 되기도 하네요 🤔

Reply:
섣부른 추상화 포스팅 한번 보시면 좋을것 같습니다~ 
[Premature Abstraction](https://injulkarnilesh.github.io/design-principles/PREMATURE_ABSTRACTION/)